hive/src/test/java/org/apache/iceberg/hive/HiveTableBaseTest.java

```diff
@@ -126,93 +116,91 @@ public void cleanup() {
+    tables.close();
+    this.tables = null;
+  }
+
+  private static HiveConf hiveConf(Configuration conf, int port) {
+    final HiveConf hiveConf = new HiveConf(conf, HiveTableBaseTest.class);
+    hiveConf.set(HiveConf.ConfVars.METASTOREURIS.varname, "thrift://localhost:" + port);
+    hiveConf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, "file:" + hiveLocalDir.getAbsolutePath());
     return hiveConf;
   }
 
-  private String getDerbyPath() {
+  private static String getDerbyPath() {
     final File metastore_db = new File(hiveLocalDir, "metastore_db");
     return metastore_db.getPath();
   }
 
-  private TServer thriftServer() throws IOException,
-          TTransportException,
-          MetaException,
-          InvocationTargetException,
-          NoSuchMethodException,
-          IllegalAccessException,
-          NoSuchFieldException {
-    final TServerSocketKeepAlive socket = new TServerSocketKeepAlive(new TServerSocket(0));
-    this.hiveConf = hiveConf(new Configuration(), socket.getServerSocket().getLocalPort());
-    HiveMetaStore.HMSHandler baseHandler = new HiveMetaStore.HMSHandler("new db based metaserver", hiveConf);
-    IHMSHandler handler = RetryingHMSHandler.getProxy(hiveConf, baseHandler, true);
-    final TTransportFactory transportFactory = new TTransportFactory();
-    final TSetIpAddressProcessor<IHMSHandler> processor = new TSetIpAddressProcessor<>(handler);
+  private static TServer thriftServer() throws Exception {
+    TServerSocket socket = new TServerSocket(0);
+    HiveTableBaseTest.hiveConf = hiveConf(new Configuration(), socket.getServerSocket().getLocalPort());
+    HiveConf serverConf = new HiveConf(hiveConf);
+    serverConf.set(HiveConf.ConfVars.METASTORECONNECTURLKEY.varname, "jdbc:derby:" + getDerbyPath() + ";create=true");
+    HiveMetaStore.HMSHandler baseHandler = new HiveMetaStore.HMSHandler("new db based metaserver", serverConf);
+    //IHMSHandler handler = RetryingHMSHandler.getProxy(serverConf, baseHandler, false);
 
     TThreadPoolServer.Args args = new TThreadPoolServer.Args(socket)
-            .processor(processor)
-            .transportFactory(transportFactory)
+            .processor(new TSetIpAddressProcessor<>(baseHandler))
+            .transportFactory(new TTransportFactory())
             .protocolFactory(new TBinaryProtocol.Factory())
             .minWorkerThreads(3)
             .maxWorkerThreads(5);
 
     return new TThreadPoolServer(args);
   }
 
-  private void setupDB(String dbURL) throws SQLException, IOException {
+  private static void setupDB(String dbURL) throws SQLException, IOException {
     Connection connection = DriverManager.getConnection(dbURL);
     ScriptRunner scriptRunner = new ScriptRunner(connection, true, true);
 
-    URL hiveSqlScript = getClass().getClassLoader().getResource("hive-schema-3.1.0.derby.sql");
+    URL hiveSqlScript = HiveTableBaseTest.class.getClassLoader().getResource("hive-schema-3.1.0.derby.sql");
```

Comment:
Do we need to change the script file as well?

Reply:
I made a small update so this one works with the 1.2.1 code.

Reply:
Would it then make sense to drop the `3.1.0` version suffix?

Reply:
I'm reluctant to because it would make the history of the file harder to follow. A rename would obscure the change because it is a whole-file delete and a whole-file add.

I'd say that the easiest thing is to keep the file name as it is. That way we know the version it was based on.

Reply:
Makes sense